src/main/java/com/github/forax/umldoc/editor/Editor.java

```diff
@@ -22,22 +22,39 @@ enum State {
     READWRITE
   }
 
+  /**
+   * The enum Extension corresponds to the type of the file.
+   * <br>Mermaid, PlantUML and Markdown.
+   */
+  enum Extension {
+    MARKDOWN,
+    PLANTUML,
+    MERMAID
+  }
+
   private final Map<String, CommandLineParser> registration;
   private final List<Package> module;
   private State state;
   private CommandLineParser parser;
+  private final Extension extension;
 
   /**
    * The constructor of Editor.
    *
    * @param registration Map, register a String with a CommandLineParser.
    * @param module List, a list of package.
    */
-  public Editor(Map<String, CommandLineParser> registration,
+  public Editor(String extension, Map<String, CommandLineParser> registration,
```

Comment:
String is a poor man object, the extension should be typed as Extension and the switch should be in the calling code